src/test/java/uk/gov/hmcts/reform/managecase/service/noc/ChallengeAnswerValidatorTest.java

```diff
@@ -0,0 +1,249 @@
+import org.mockito.MockitoAnnotations;
+import uk.gov.hmcts.reform.managecase.client.datastore.model.elasticsearch.SearchResultViewItem;
+import uk.gov.hmcts.reform.managecase.client.definitionstore.model.ChallengeQuestion;
+import uk.gov.hmcts.reform.managecase.client.definitionstore.model.ChallengeQuestionsResult;
+import uk.gov.hmcts.reform.managecase.client.definitionstore.model.FieldType;
+import uk.gov.hmcts.reform.managecase.domain.SubmittedChallengeAnswer;
+
+import javax.validation.ValidationException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SuppressWarnings({"PMD.JUnitAssertionsShouldIncludeMessage", "PMD.DataflowAnomalyAnalysis",
+    "PMD.TooManyMethods", "PMD.UseConcurrentHashMap"})
+class ChallengeAnswerValidatorTest {
+
+    public static final String QUESTION_ID_1 = "QuestionId1";
+    public static final String QUESTION_ID_2 = "QuestionId2";
+    public static final String QUESTION_ID_3 = "QuestionId3";
+    public static final String TEXT = "Text";
+
+    @InjectMocks
+    private ChallengeAnswerValidator challengeAnswerValidator;
+
+    private List<SubmittedChallengeAnswer> answers;
+    private SearchResultViewItem caseSearchResult;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeEach
+    void setUp() throws JsonProcessingException {
+        MockitoAnnotations.initMocks(this);
+
+        answers = new ArrayList<>();
+        answers.add(new SubmittedChallengeAnswer(QUESTION_ID_1, " T-e xt'VALUE"));
+        answers.add(new SubmittedChallengeAnswer(QUESTION_ID_2, "67890"));
+        answers.add(new SubmittedChallengeAnswer(QUESTION_ID_3, "1985-07-25"));
+
+        caseSearchResult = createCase();
+    }
+
+    @Test
+    void shouldSuccessfullyIdentifyMatchingCaseRole() {
+        List<ChallengeQuestion> challengeQuestions = new ArrayList<>();
+        challengeQuestions.add(
+            challengeQuestion(QUESTION_ID_1, "${TextField}|${SomeOtherField}:[Claimant],${AnotherField}:[Defendant]",
+                fieldType(TEXT))
+        );
+        challengeQuestions.add(
+            challengeQuestion(QUESTION_ID_2, "${ComplexField.ComplexNestedField.NestedNumberField}:[Claimant],"
+                + "${AnotherField}:[Defendant]", fieldType("Number"))
+        );
+        challengeQuestions.add(
+            challengeQuestion(QUESTION_ID_3, "${AnotherField}:[Defendant],${NonExistingField}|${DateField}:[Claimant]",
+                fieldType("Date"))
+        );
+        ChallengeQuestionsResult challengeQuestionsResult = new ChallengeQuestionsResult(challengeQuestions);
+
+        String result = challengeAnswerValidator
+            .getMatchingCaseRole(challengeQuestionsResult, answers, caseSearchResult);
+
+        assertEquals("[Claimant]", result);
```

Comment:
Should this use the matchers used elsewhere in unit tests

`assertThat(result).isEqualTo("[Claimant]")` ?